app/controllers/missions_controller.rb

```diff
@@ -1,47 +1,49 @@
 class MissionsController < ApplicationController
-  VALID_MISSION_PARAMS = %w(name description)
+  VALID_MISSION_PARAMS = %w(name description public)
 
-  before_action :set_mission, only: [:show, :edit, :update, :destroy]
+  before_action(:set_mission, {
+    only: [:show, :edit, :update, :destroy]
+  })
+
+  before_action({ only: [:show, :edit] }) do
+    require_user! unless @mission.public?
+  end
 
-  # GET /missions
-  # GET /missions.json
   def index
-    @missions = Mission.all
+    @missions = Mission.where({ public: true })
   end
 
-  # GET /missions/1
-  # GET /missions/1.json
   def show
+    authorize!(:write, @mission)
   end
 
-  # GET /missions/new
   def new
     @mission = Mission.new
   end
 
-  # GET /missions/1/edit
   def edit
+    authorize!(:write, @mission)
   end
 
-  # POST /missions
-  # POST /missions.json
   def create
     @mission = Mission.new(mission_params)
+    if !@mission.public? && !current_user
```

Comment:
I originally had that aliased but thought it would cause confusion as there's no difference right now -- should I just rename the method?